### PR TITLE
ClusterSharding: increasing retry interval for registration call, #25191

### PR DESCRIPTION
**Pull request by patriknw**
Completing what was started in https://github.com/akka/akka/pull/25228

Refs #25191

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala [hunk 1]
```diff
@@ -758,6 +771,25 @@ private[akka] class ShardRegion(
...
+    register()
+    scheduleNextRegistration()
+  }
+
+  def scheduleNextRegistration(): Unit = {
+    if (nextRegistrationDelay < retryInterval) {
+      timers.startSingleTimer(RegisterRetry, RegisterRetry, nextRegistrationDelay)
+      // exponentially increasing retry interval until reaching the normal retryInterval
+      nextRegistrationDelay *= 2
+    }
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: Shouldn't it `startSingleTimer` also when reached max delay?

Comment (patriknw) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: nope, then the normal `retryInterval` is used

Comment (patriknw) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala]: see `timers.startPeriodicTimer(Retry, Retry, retryInterval)` in preStart

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
merging with 1 review since I also reviewed it (not implemented by me)

**Pull request merged**